### PR TITLE
[JavaScript] Prepare the track

### PR DESCRIPTION
**Pull request by SleeplessByte**
- Convert existing files to new repository structure
- Write a Concept Exercise implementation guide
- List out key Concepts for your language

I hope this is right...

### CODE REVIEW

File: languages/javascript/README.md [hunk 1]
```diff
@@ -51,5 +53,5 @@ These extra steps will make your track better, but are optional.
 
 ### Analyzer
 
-- [ ] Build Analyzer
+- [X] Build Analyzer
 - [ ] Deploy Analyzer
```
Comment (iHiD) [inline languages/javascript/README.md]: I believe this can also be ticked?

File: languages/javascript/exercises/concept/README.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# JavaScript concept exercises
+
+The concept exercises in the JavaScript track are a work-in-progress and are based on this [list of concepts][reference-shared]. Important _types_ of concepts to target are things that only exist in object-oriented programming (for people coming from non-oop languages), functions as a first class citizen (for people coming from non-functional languages), and JavaScript specifics:
```
Comment (iHiD) [inline languages/javascript/exercises/concept/README.md]: We're treating "Concept Exercises" and "Concepts" as _proper nouns_ within the context of Exercism, so they should be capitalized.

Comment (SleeplessByte) [reply languages/javascript/exercises/concept/README.md]: Oki!

File: languages/javascript/exercises/shared/.docs/debug.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Debug
+
+When a test fails, a message is displayed describing what went wrong and for which input. You can also use the fact that any `console` output will be shown too. You can write to the console using:
+
+```javascript
+console.log('Debug message')
```
Comment (iHiD) [inline languages/javascript/exercises/shared/.docs/debug.md]: OOI, is this implemented in the Test Runner?

**Comment by iHiD** [comment]
I've briefly scanned this and it looks awesome. I'll let @ErikSchierboom do a proper review tomorrow, but unless he notices any fundamental issues, I suggest we merge it and them can just keep tweaking it moving forward. I think generally having **some** stuff on master is better than having **perfect** stuff on master, with the exception of the actual Concept Exercises.

@SleeplessByte Thank you :)

File: languages/javascript/exercises/shared/.docs/debug.md (hunk 3 above)
Comment (SleeplessByte) [reply languages/javascript/exercises/shared/.docs/debug.md]: It is, but it isn't working quite right yet XD

File: languages/javascript/README.md (hunk 1 above)
Comment (SleeplessByte) [reply languages/javascript/README.md]: I guess 😄

File: languages/javascript/docs/implementing-a-concept-exercise.md [hunk 4]
```diff
@@ -1,5 +1,141 @@
-# How to implement an JavaScript concept exercise
+# How to implement a JavaScript concept exercise
 
-TODO: describe how to implement a concept exercise for the JavaScript track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes how to implement a concept exercise for the JavaSCript track. As this document is generic, the following placeholders are used:
```
Comment (ErikSchierboom) [inline languages/javascript/docs/implementing-a-concept-exercise.md]:
```suggestion
This document describes how to implement a concept exercise for the JavaScript track. As this document is generic, the following placeholders are used:
```

File: languages/javascript/docs/implementing-a-concept-exercise.md [hunk 5]
```diff
@@ -1,5 +1,141 @@
...
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<slug>`: the name of the exercise in snake_case (e.g. `anonymous-methods`).
```
Comment (ErikSchierboom) [inline languages/javascript/docs/implementing-a-concept-exercise.md]: I like this list of placeholders!

**Comment by iHiD** [comment]
Whoop!

**Pull request merged**